src/main/java/se/kth/sda/skeleton/posts/PostController.java

```diff
@@ -1,8 +1,57 @@
 package se.kth.sda.skeleton.posts;
 
-/*
-    @TODO create the methods needed to implement the API.
-    Don't forget to add necessary annotations.
- */
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import se.kth.sda.skeleton.posts.exeption.*;
+import se.kth.sda.skeleton.posts.PostService;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RequestMapping("/posts")
+@RestController
 public class PostController {
+    PostRepository postRepository;
+    PostService postService;
+
+    @Autowired
+    public PostController(PostRepository postRepository, PostService postService) {
+        this.postRepository = postRepository;
+        this.postService = postService;
+    }
+    /* create a new post */
+    @PostMapping("/")
+    public ResponseEntity<Post> createPost(@Valid @RequestBody Post post) {
+        return ResponseEntity.status(HttpStatus.CREATED)
+                .body(postService.createPost(post));
+    }
+
+    /* return all posts*/
+    @GetMapping("/")
+    public ResponseEntity <List<Post>>listAllPost() {
+        return ResponseEntity.ok(postService.listAllPosts());
+    }
+
+    /* return a specific post based on the provided id */
+    @GetMapping("/{id}")
+    public ResponseEntity<Post> getPostById(@PathVariable Long id){
+        Post post = postService.getPostById(id);
+        return ResponseEntity.ok(post);
+    }
+
+    /* return should update the post by id */
+    @PutMapping("/{id}")
+    public ResponseEntity<Post> updatePostById(@PathVariable Long id, @RequestBody Post updatePostById){
+        Post post = postService.updatePost(id, updatePostById);
+        return ResponseEntity.ok(post);
```

Comment:
same as previous comment regarding 2 methods with similar name:
`updatePostById`
`updatePost`
even if here it makes more sense actually

Reply:
Ok, we are now changing the method names in PostController to match the ones in PostService